java-vtl-script/src/main/java/no/ssb/vtl/script/operations/join/AbstractJoinOperation.java

```diff
@@ -390,6 +395,25 @@ protected DataStructure computeDataStructure() {
         return newDataStructure.build();
     }
 
+    /** Checks if component is unique among other datasets  */
+    private boolean componentIsUnique(String datasetName, Map.Entry<String, Component> componentEntry) {
```

Comment:
why do you need to check the type and role of the components here?

Reply:
It is defined by the VTL spec. Se lines 1811-1812 for join, or 3879 for union in VTL 1.1 Part 2 Reference Manual.

Reply:
If the on clause is specified then the join is possibly defined on a subset of the common Identifier Components of the Datasets.

If the Datasets have common Identifier Components (i.e. with identical names, data type and values domain) that are not specified in the on clause then it is mandatory to refer to those Identifier Components by specifying both the Dataset name and the measure name. 

For example, if ds1 and ds2 have some common Identifier Components d1, d2 and d3, the following expression:
`[ ds1,ds2 on d1, d2 ]`
 returns a Dataset with the following Identifier Components:
* d1
* d2
* 'ds1.d3'
* 'ds2.d3'

the Identifier Components can be renamed using the rename clause:
`[ ds1,ds2 on d1, d2 ] { rename 'ds1.d3' to new1, 'ds2_d3' to new2 }`

In the final result of join expression the common Identifier Components that are not listed in the on clause and have not been renamed are automatically renamed by replacing the "." with an underscore "_". The same applies when the working Dataset is the input for a filter that uses a datapoint (horizontal) ruleset.

Reply:
The spec is confusing. They are talking about the subset of the common identifiers (ie. same name, type and role) in the context of a join with `on`.
In this scenario some of the common identifiers are "downgraded" to regular measures since we are not joining on them. 

This logic should apply for components with same names regardless of its role and type. The subset of identifier we can join on is handled in subclasses of the AbstractJoinOperation.

Reply:
Actually, lines 1797-1798:
> If the Datasets have common measures or attributes (i.e. with identical names) then [...]
define "common" for measures and attributes as having only identical names.
So do we have 2 definitions of "common" or is this a mistake?

Reply:
But I see your point that if the name is the same, the measures/attributes should get a prefix and we don't need to check the rest (role, type).